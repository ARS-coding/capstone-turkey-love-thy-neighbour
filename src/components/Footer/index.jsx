import React from "react";

import { Link } from "react-router-dom";

import { Container, Row, Col } from "react-bootstrap";

import { ReactComponent as TwitterLogo } from "../../images/twitter-logo.svg";
import { ReactComponent as InstagramLogo } from "../../images/instagram-logo.svg";
import { ReactComponent as FacebookLogo } from "../../images/facebook-logo.svg";

import "./index.scss";

function Footer() {
  return (
    <footer className="footer container-fluid d-flex align-items-center pt-2">
      <Container>
        <Row className="footer-content-row w-100 h-100">
          <Col
            xs={12}
            sm={12}
            md={4}
            className="d-flex justify-content-center justify-content-sm-center justify-content-md-start align-items-center"
          >
            <Link to="our-team">Our Team</Link>
            <Link to="contact-us" className="mx-3">
              Contact Us
            </Link>
            <Link to="faq">FAQ</Link>
          </Col>
          <Col
            xs={12}
            sm={12}
            md={4}
            className="d-flex justify-content-center align-items-center my-2 my-sm-2 my-sm-0"
          >
            <p>Copyright Â© Re:Coded 2021</p>
          </Col>
          <Col
            xs={12}
            sm={12}
            md={4}
            className="social-media-logos d-flex justify-content-center justify-content-sm-center justify-content-md-end align-items-center"
          >
            <a
              href="https://twitter.com"
              target="_blank"
              rel="external noreferrer"
            >
              <TwitterLogo />
            </a>
            <a
              href="https://www.instagram.com"
              target="_blank"
              rel="external noreferrer"
            >
              <InstagramLogo />
            </a>
            <a
              href="https://www.facebook.com"
              target="_blank"
              rel="external noreferrer"
            >
              <FacebookLogo />
            </a>
          </Col>
        </Row>
      </Container>
    </footer>
  );
}

export default Footer;
