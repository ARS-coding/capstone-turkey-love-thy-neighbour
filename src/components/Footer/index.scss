@import "../../style/variables";

.footer {
  min-height: 60px;
  background-color: $dark-blue;

  font-family: "Poppins";

  p {
    margin: 0;
    font-size: 10px;
    color: $white;
  }

  a {
    font-size: 16px;
    text-decoration: none;
    color: rgba(255, 255, 255, 0.55);
    transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out,
      border-color 0.15s ease-in-out;

    &:hover {
      color: rgba(255, 255, 255, 0.75);
    }
  }

  .footer-content-row {
    margin: 0;

    .social-media-logos {
      svg {
        transition: all ease-in-out 300ms;
      }
      svg:hover {
        filter: brightness(90%) contrast(10%);
        transition: all ease-in-out 300ms;
      }
    }

    & > div {
      padding: 0;
    }
  }
}
