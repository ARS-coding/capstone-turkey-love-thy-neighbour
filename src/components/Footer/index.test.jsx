import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
import Footer from ".";

describe("Footer Section", () =>
  it("renders Footer section correctly", () => {
    const tree = renderer
      .create(
        <Router>
          <Footer />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  }));
