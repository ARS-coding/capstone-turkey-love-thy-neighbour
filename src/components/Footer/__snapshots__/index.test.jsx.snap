// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Section renders Footer section correctly 1`] = `
<footer
  className="footer container-fluid d-flex align-items-center pt-2"
>
  <div
    className="container"
  >
    <div
      className="footer-content-row w-100 h-100 row"
    >
      <div
        className="mt-3 mt-sm-3 mt-md-0 d-flex justify-content-center justify-content-sm-center justify-content-md-start align-items-center col-md-4 col-sm-12 col-12"
      >
        <a
          href="/our-team"
          onClick={[Function]}
        >
          Our Team
        </a>
        <a
          className="mx-3"
          href="/contact-us"
          onClick={[Function]}
        >
          Contact Us
        </a>
        <a
          href="/faq"
          onClick={[Function]}
        >
          FAQ
        </a>
      </div>
      <div
        className="d-flex justify-content-center align-items-center my-2 my-sm-2 my-sm-0 col-md-4 col-sm-12 col-12"
      >
        <p>
          Copyright Â© Re:Coded 2021
        </p>
      </div>
      <div
        className="social-media-logos d-flex justify-content-center justify-content-sm-center justify-content-md-end align-items-center col-md-4 col-sm-12 col-12"
      >
        <a
          href="https://twitter.com"
          rel="external noreferrer"
          target="_blank"
        >
          <svg>
            twitter-logo.svg
          </svg>
        </a>
        <a
          href="https://www.instagram.com"
          rel="external noreferrer"
          target="_blank"
        >
          <svg>
            instagram-logo.svg
          </svg>
        </a>
        <a
          href="https://www.facebook.com"
          rel="external noreferrer"
          target="_blank"
        >
          <svg>
            facebook-logo.svg
          </svg>
        </a>
      </div>
    </div>
  </div>
</footer>
`;
